networkit/cpp/planarity/LeftRightPlanarityCheck.cpp

```diff
@@ -0,0 +1,282 @@
+            ref[edge] = highestReturnEdgeLeft;
+        } else {
+            ref[edge] = highestReturnEdgeRight;
+        }
+    }
+}
+
+bool LeftRightPlanarityCheck::dfsTesting(node startNode) {
+    std::stack<node> dfsStack;
+    dfsStack.emplace(startNode);
+    std::unordered_map<node, decltype(dfsGraph.neighborRange(startNode).begin())> neighborIterators;
+    std::unordered_set<Edge> preprocessedEdges;
+
+    auto processNeighborEdges = [&](node currentNode, bool &callRemoveBackEdges) -> bool {
+        auto &neighborIterator = neighborIterators[currentNode];
+        while (neighborIterator != dfsGraph.neighborRange(currentNode).end()) {
+            const node neighbor = *neighborIterator;
+            const Edge currentEdge(currentNode, neighbor);
+
+            if (!preprocessedEdges.contains(currentEdge)) {
+                stackBottom[currentEdge] = stack.empty() ? NoneConflictPair : stack.top();
+                if (currentEdge == parentEdges[neighbor]) {
+                    dfsStack.emplace(currentNode);
+                    dfsStack.emplace(neighbor);
+                    preprocessedEdges.insert(currentEdge);
+                    callRemoveBackEdges = false;
+                    return true; // Indicate further processing needed
+                }
+
+                lowestPointEdge[currentEdge] = currentEdge;
+                stack.emplace(Interval{}, Interval(currentEdge, currentEdge));
+            }
+
+            if (auto currentEdgeIterator = lowestPoint.find(currentEdge);
+                currentEdgeIterator != lowestPoint.end()
+                && currentEdgeIterator->second < heights[currentNode]) {
+
+                if (neighbor == *dfsGraph.neighborRange(currentNode).begin()) {
+                    lowestPointEdge[parentEdges[currentNode]] = lowestPointEdge[currentEdge];
+                } else {
+                    if (!applyConstraints(currentEdge, parentEdges[currentNode])) {
+                        return false;
+                    }
+                }
+            }
+            ++neighborIterator;
+        }
+        return true;
+    };
+
+    // Main DFS loop
+    do {
+        const node currentNode = dfsStack.top();
+        dfsStack.pop();
+        const Edge parentEdge = parentEdges[currentNode];
+        bool callRemoveBackEdges{true};
+
+        if (auto it = neighborIterators.find(currentNode); it == neighborIterators.end()) {
+            neighborIterators[currentNode] = dfsGraph.neighborRange(currentNode).begin();
+        }
+
+        if (!processNeighborEdges(currentNode, callRemoveBackEdges)) {
+            return false;
+        }
+
+        if (callRemoveBackEdges) {
+            if (parentEdge != noneEdge) {
+                removeBackEdges(parentEdge);
+            }
+        }
+    } while (!dfsStack.empty());
+
+    return true;
+}
+
+void LeftRightPlanarityCheck::dfsOrientation(node startNode) {
+    std::stack<node> dfsStack;
+    dfsStack.emplace(startNode);
```

Comment:
`emplace` is useful for constructing objects inplace, but using it for primitive type is equivalent to `push`. So for this stack I'd suggest to use `push`, see more here https://abseil.io/tips/112.